### ISSUE TITLE
instaling sage

### ISSUE THREAD

**Issue by thirny** [cschwan/sage-on-gentoo#804]
emerge -va sage" - command is not working? is it depracated step of installing?

**Comment by Zeggpold87** [comment]
The README.rst file still says to use emerge -va sage.

**Comment by kiwifb** [comment]
Yes, indeed and it will not work anymore, because I have replaced sage with sagemath-standard. I should give myself a slap for that.
Although "sage" may make a return sooner rather than later.

**Comment by tobiasdiez** [comment]
I would recommend to use `sage` (or perhaps `sagemath`) for the meson build (the pyproject.toml file in the root). At least this naming "convention" was proposed in https://github.com/sagemath/sage/pull/39548 for the pypi distributions.

**Comment by kiwifb** [comment]
> I would recommend to use `sage` (or perhaps `sagemath`) for the meson build (the pyproject.toml file in the root). At least this naming "convention" was proposed in [sagemath/sage#39548](https://github.com/sagemath/sage/pull/39548) for the pypi distributions.

That was my thinking. But it would be nice to have clarity of naming. I am still not happy about optional package handling in the meson build. The modular approach was better (I do not care so much about the rest of the modularization work), but by far the best approach in my opinion is the outright separation as performed for giac.

**Comment by tobiasdiez** [comment]
> I am still not happy about optional package handling in the meson build. The modular approach was better (I do not care so much about the rest of the modularization work), but by far the best approach in my opinion is the outright separation as performed for giac.

What is it in particular that you dislike about the meson build? It's design is definitely not set in stone, and we can change it to accommodate needs of distro managers.

**Comment by kiwifb** [comment]
Sorry for taking so much time fixing this. My real work has been "interesting" over the last 3 months. And I am just taking a breather before diving in again for at least a couple of months.

@tobiasdiez I think I have stated some of my issues in the sagemath repository already. Some points in order
* Replacing setuptools by meson is welcome - and your work on this is very much appreciated.
* timing has not been great and it is nothing anyone can help with. I have invested some time in the split packages and now I will probably need to roll it back :(
* Nitpick 1: I have always considered the top level directory as being for sage the distribution and sage proper being in `src`. I am guessing you may want to replace the whole sage-the-distro build system in time. I would have preferred the top meson directory to be in `src`. I guess so long as a sdist produced by meson for sage does not include sage-the-distro stuff, I am good. But I will have to change all my patches which are applied relative to `src`. Annoying but sed can do that job at scale.
* Nitpick 2: one of the real good thing about the modularisation scheme is that it split the optional packages (mcqd, sirrocco, bliss...) eliminating the automagic detection of stuff present at compilation time. The modularization make this fine grained and you can add or remove them after sage proper. Your work goes back to automagic detection which in Gentoo is considered bad form. That being said, the modularization work insisted on keeping everything in one git tree instead of splitting things in different trees. There are pros and cons to that. But a split like the one operated for giac/sagemath-giac is cleaner.

Out of these, nitpick 2 is my biggest annoyance. I really want the optional packages to be separate packages - and they can use meson too.

**Comment by tobiasdiez** [comment]
Thanks for the detailed feedback! Very much appreciated.

Two quick answers:
> But I will have to change all my patches which are applied relative to src

Hopefully you don't need any patches for sagelib (anything in src) with meson. The env variable handling (sage.env) definitely should work better now. Not sure if you still need your patches for mpmath. But maybe they can also be directly incorporated upstream in sage?

> But a split like the one operated for giac/sagemath-giac is cleaner.

I definitely agree. The optional stuff should (mostly) go into separate packages. It's just a question who has time to do this migration. Note that the meson build is actually very helpful in this direction: the new giac package just copy-pasted the existing meson build file from sage.

On the other hand, making a few more dependencies optional should hopefully make it easier for downstream packaging to provide up-to-date sage (perhaps with a few optional features missing; which is still better than no sage at all in say ubuntu or fedora).

Let me know if there is something we can improve in the meson build to make your life as a downstream maintainer easier.

**Comment by kiwifb** [comment]
I actually found the time to package a meson build. It allowed to see where we are at with it. Ported all my patch and yes, I had to make new bits for mpmath for the meson build (https://github.com/cschwan/sage-on-gentoo/blob/vbraun/sci-mathematics/sagemath/files/mpmath-10.6.patch this is on my development branch). Definitely need to make a new branch for upcoming mpmath 1.4 (release when?).

What I found from the experience was that only a minimal "sage" start up script was installed. It was insufficient to make a documentation package after that. The doc build stops when "sage --advanced" returns an error.

**Comment by tobiasdiez** [comment]
> I actually found the time to package a meson build.

Awesome! Thanks.

> What I found from the experience was that only a minimal "sage" start up script was installed. 

This is by design. We took the opportunity to cleanup the interface and remove most of the stuff that doesn't make sense for non-developers and if you are not using sage-the-distro. A few useful commands may still be missing though.

> It was insufficient to make a documentation package after that. The doc build stops when "sage --advanced" returns an error.

The doc build is next on my todo list, see https://github.com/sagemath/sage/pull/39973.
